src/samples/agents/sample-agent/index.ts

```diff
@@ -0,0 +1,182 @@
+import express from "express";
+import { v4 as uuidv4 } from 'uuid'; // For generating unique IDs
+
+import {
+  AgentCard,
+  Task,
+  TaskState,
+  TaskStatusUpdateEvent,
+  Message
+} from "../../../index.js";
+import {
+  InMemoryTaskStore,
+  TaskStore,
+  AgentExecutor,
+  RequestContext,
+  ExecutionEventBus,
+  DefaultRequestHandler
+} from "../../../server/index.js";
+import { A2AExpressApp } from "../../../server/express/index.js";
+
+/**
+ * SampleAgentExecutor implements the agent's core logic.
+ */
+class SampleAgentExecutor implements AgentExecutor {
+  private cancelledTasks = new Set<string>();
+
+  public cancelTask = async (
+    taskId: string,
+    eventBus: ExecutionEventBus,
+  ): Promise<void> => { };
+
+  async execute(
+    requestContext: RequestContext,
+    eventBus: ExecutionEventBus
+  ): Promise<void> {
+    const userMessage = requestContext.userMessage;
+    const existingTask = requestContext.task;
+
+    // Determine IDs for the task and context
+    const taskId = requestContext.taskId;
+    const contextId = requestContext.contextId;
+
+    console.log(
+      `[SampleAgentExecutor] Processing message ${userMessage.messageId} for task ${taskId} (context: ${contextId})`
+    );
+
+    // 1. Publish initial Task event if it's a new task
+    if (!existingTask) {
+      const initialTask: Task = {
+        kind: 'task',
+        id: taskId,
+        contextId: contextId,
+        status: {
+          state: 'submitted',
+          timestamp: new Date().toISOString(),
+        },
+        history: [userMessage], // Start history with the current user message
+        metadata: userMessage.metadata, // Carry over metadata from message if any
+      };
+      eventBus.publish(initialTask);
+    }
+
+    // 2. Publish "working" status update
+    const workingStatusUpdate: TaskStatusUpdateEvent = {
+      kind: 'status-update',
+      taskId: taskId,
+      contextId: contextId,
+      status: {
+        state: 'working',
+        message: {
+          kind: 'message',
+          role: 'agent',
+          messageId: uuidv4(),
+          parts: [{ kind: 'text', text: 'Processing your question' }],
+          taskId: taskId,
+          contextId: contextId,
+        },
+        timestamp: new Date().toISOString(),
+      },
+      final: false,
+    };
+    eventBus.publish(workingStatusUpdate);
+
+    // 3. Publish final task status update
+    const agentReplyText = "The sample agent correctly processed your request.";
```

Comment:
Can we have a few options here like mentioned in https://github.com/a2aproject/a2a-js/issues/133#issuecomment-3404840133?